cbor2/decoder.py

```diff
@@ -90,20 +94,24 @@ def decode_map(decoder, subtype, shareable_index=None):
     if length is None:
         # Indefinite length
         while True:
-            key = decoder.decode()
+            with decoder.key_decoder() as decode:
+                key = decode()
             if key is break_marker:
                 break
             else:
                 value = decoder.decode()
                 dictionary[key] = value
     else:
         for _ in xrange(length):
-            key = decoder.decode()
+            with decoder.key_decoder() as decode:
+                key = decode()
             value = decoder.decode()
             dictionary[key] = value
 
     if decoder.object_hook:
         return decoder.object_hook(decoder, dictionary)
+    elif decoder._hashable:
```

Comment:
See the comment further down.

Reply:
Why is `_hashable` private but `key_decoder` is public? That makes no sense. If someone wants to implement their own decoder, it would make sense for them to want to read the state of `hashable`, yes?